### PR TITLE
Deployment & Test Tasks

### PR DESCRIPTION
**Pull request by kmkumaran**
Porting Deployment and Test tasks into github.

Tasks added: 
Chef, CopyFilesToAzureBlob, CopyFilesToMachines, DeployAzureResourceGroup, Docker

Tasks Modified:
DeployVisualstudioTestAgent, RunPowerShellOntargetMachines, RunVisualstudioTestAgent

### CODE REVIEW

File: Tasks/Chef/task.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
+﻿{
+    "id": "B719DB6C-40A2-4F43-9AFF-827825BAECAE",
+    "name": "Chef",
+    "friendlyName": "Chef",
+    "description": "Deploy to Chef environments by editing environment attributes",
+    "category": "Deploy",
+    "visibility": [
```
Comment (praneethy91) [inline Tasks/Chef/task.json]: Why are we having visibility in  two different places? is this a merge issue which we didn't catch in subraman's PR?

File: Tasks/CopyFilesToMachines/task.json [hunk 2]
```diff
@@ -0,0 +1,106 @@
+﻿{
+    "id": "731004D4-1D66-4F70-8C05-638018B22210",
+    "name": "CopyFilesToMachines",
+    "friendlyName": "Copy Files To Machines",
+    "description": "Copy files to remote machine(s)",
+    "category": "Deploy",
+    "visibility": [
```
Comment (praneethy91) [inline Tasks/CopyFilesToMachines/task.json]: We are having visibility in two different places here too.

File: Tasks/DeployAzureResourceGroup/task.json [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    "friendlyName": "Deploy Azure Resource Group",
+    "description": "CSM deployment to azure",
+    "category": "Deploy",
+    "visibility": [
+                "Preview",
+                "Build",
+                "Release"
+                  ],
+    "author": "Microsoft Corporation",
+    "visibility": [
```
Comment (praneethy91) [inline Tasks/DeployAzureResourceGroup/task.json]: Here also we are having in two different places, the visibiilty.

File: Tasks/CopyFilesToAzureBlob/task.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            "required": true,
+            "helpMarkDown": "Azure subscription to configure before running PowerShell"
+        },
+        {
+            "name": "SourcePath",
+            "type": "string",
+            "label": "Source",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Location of the source folder or file. Variables like $(Build.Repository.LocalPath) can be used for binaries in the Automation Agent. For example, $(Build.Repository.LocalPath)\\BudgetIT\\Web."
```
Comment (omeshp) [inline Tasks/CopyFilesToAzureBlob/task.json]: Variables like $(Build.Repository.LocalPath) will not work in Release context, we should change the text to something generic which applies to both.

File: Tasks/RunDistributedTests/task.json [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
     "id": "D353D6A2-E361-4A8F-8D8C-123BEBB71028",
-    "name": "RunDistributedTests",
-    "friendlyName": "Run Tests using Test Agent",
-    "description": "Run tests distributedly using multiple test agents in a lab environment. This task is under Preview, please contact devops_tools@microsoft.com to enable this task fully on the backend for your account. Please refer http://aka.ms/ci_cd for additional guidance.",
+    "name": "RunVisualStudioTestsusingTestAgent",
+    "friendlyName": "Run Visual Studio Tests using Test Agent",
+    "description": "Run tests distributedly using multiple test agents in a lab machine group.",
     "category": "Test",
     "visibility": [
                 "Preview",
```
Comment (acesiddhu) [inline Tasks/RunDistributedTests/task.json]: This should have build and preview. (No RM for now)

File: Tasks/DeployTestAgent/task.json [hunk 6]
```diff
@@ -1,8 +1,8 @@
 ﻿{
     "id": "52A38A6A-1517-41D7-96CC-73EE0C60D2B6",
-    "name": "DeployTestAgent",
+    "name": "DeployVisualStudioTestAgent",
     "friendlyName": "Deploy Visual Studio Test Agent",
-    "description": "Deploy and configure Test Agent to run tests on a lab environment. This task is under Preview, please contact devops_tools@microsoft.com to enable this task fully on the backend for your account. Please refer http://aka.ms/ci_cd for additional guidance.",
+    "description": "Deploy and configure Test Agent to run tests on a lab machine group.",
     "category": "Test",
     "visibility": [
                 "Preview",
```
Comment (acesiddhu) [inline Tasks/DeployTestAgent/task.json]: This should have build and preview. (No RM for now)

File: Tasks/RunPowerShellOnTargetMachines/task.json [hunk 7]
```diff
@@ -1,27 +1,29 @@
...
-    "description": "Deploy to TFS Environment.  Executes PowerShell commands remotely on machine(s). This task is under Preview, please contact devops_tools@microsoft.com to enable this task fully on the backend for your account. Please refer http://aka.ms/ci_cd for additional guidance.",
+    "name": "RunPowerShellOnTargetMachines",
+    "friendlyName": "Run PowerShell on Target Machines",
+    "description": "Run PowerShell on Target Machines.  Executes PowerShell commands remotely on machine(s). This task is under Preview, please contact devops_tools@microsoft.com to enable this task fully on the backend for your account. Please refer http://aka.ms/ci_cd for additional guidance.",
     "category": "Deploy",
     "visibility": [
                 "Preview",
-                "Build"
+                "Build",
+                "Release"
```
Comment (acesiddhu) [inline Tasks/RunPowerShellOnTargetMachines/task.json]: i think when we have both build and RM. we just need to specify Preview. only for dta task we need preview and build

File: Tasks/CopyFilesToAzureBlob/task.json (hunk 4 above)
Comment (chrisrpatterson) [inline Tasks/CopyFilesToAzureBlob/task.json]: Please don't use specific variables in descriptions here.  If the type of the input is filePath you can use text like "relative path in the default working directory" Other wise just describe what it is.

File: Tasks/DeployAzureResourceGroup/task.json (hunk 3 above)
Comment (kmkumaran) [inline Tasks/DeployAzureResourceGroup/task.json]: I have taken care of duplicate visibility in jsons.

File: Tasks/CopyFilesToAzureBlob/task.json (hunk 4 above)
Comment (kmkumaran) [inline Tasks/CopyFilesToAzureBlob/task.json]: Thanks Omesh/Chris. I have removed the $ variables that are specific to Build from Help strings.

File: Tasks/RunDistributedTests/task.json (hunk 5 above)
Comment (kmkumaran) [inline Tasks/RunDistributedTests/task.json]: Jeff - Thanks for pointing out on azureConnection. These changes came into master after I pulled the branch. I did a merge to pick those changes. With regard to csproj, I don't see any csproj. I double checked.

Siddhartha - This change does not include modifications to visibility for Test tasks. They were already present in master. The edits are only to Names as you can see in the diff. I checked the visibility, it does not have Release.

**Comment by kmkumaran** [comment]
Have published addressing the comments. Please review.

**Pull request merged**